src/Components/CustomerProfile.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from "react";
+
+const CustomerProfile = ({ id }) => {
+  const [currentCustomer, setCurrentCustomer] = useState({});
+
+  useEffect(() => {
+    fetch(`https://cyf-react.glitch.me/customers/${id}`)
+      .then(res => res.json())
+      .then(res => setCurrentCustomer(res));
+  }, [id]);
+  const customerInfo = currentCustomer => {
+    if (id) {
+      let customerInfoSet = new Set(["id", "email", "vip", "phoneNumber"]);
```

Comment:
There is no need to create a component inside component, declare a `customerInfoSet` variable, then map over `currentCustomer` and so on. There is a software design principle called KISS and it basically means "Keep it simple, stupid". Following this principle, I don't think this component needs so much logic. Lines 11 to 39 could be simplified to the following code:
```
  return Object.keys(currentCustomer).length ? (
    <div className="alert alert-primary">
      Customer {id} Profile:
      <ul>
        <li>ID: {currentCustomer.email}</li>
        <li>EMAIL: {currentCustomer.email}</li>
        <li>VIP: {currentCustomer.vip ? "Yes" : "No"}</li>
        <li>PHONE NUMBER: {currentCustomer.phoneNumber}</li>
      </ul>
    </div>
  ) : (
    <div>Loading...</div>
  );
```
It's 29 lines of code vs 13 lines of code🙂